src/main/sensors/battery.c

```diff
@@ -159,8 +159,21 @@ void batteryUpdateVoltage(timeUs_t currentTimeUs)
             break;
     }
 
+    if (!voltageMeter.isVoltageStable && cmp32(millis(), voltageMeter.prevDisplayTime) >= PREV_DISPLAY_FILTERED_TIME_DIFF) {
+        if ((voltageState == BATTERY_NOT_PRESENT || voltageState == BATTERY_INIT) && isVoltageFromBattery()) {
+            voltageMeter.isVoltageStable = abs(voltageMeter.prevDisplayFiltered - voltageMeter.displayFiltered) <= VBAT_STABLE_MAX_DELTA;
+        }
+
+        voltageMeter.prevDisplayFiltered = voltageMeter.displayFiltered;
+        voltageMeter.prevDisplayTime = millis();
+    }
+
     DEBUG_SET(DEBUG_BATTERY, 0, voltageMeter.unfiltered);
     DEBUG_SET(DEBUG_BATTERY, 1, voltageMeter.displayFiltered);
+    DEBUG_SET(DEBUG_BATTERY, 4, voltageMeter.isVoltageStable ? 1 : 0);
+    DEBUG_SET(DEBUG_BATTERY, 5, isVoltageFromBattery() ? 1 : 0);
+    DEBUG_SET(DEBUG_BATTERY, 6, abs(voltageMeter.prevDisplayFiltered - voltageMeter.displayFiltered));
```

Comment:
Making Debug 6 to log only `prevDisplayFiltered` shows it stepping upwards and then holding the final value, making it easier to visualise the ultimate battery voltage estimate.